WordPress/gutenberg
# Fix labeling of the navigation links in the list view.

Fixes https://github.com/WordPress/gutenberg/issues/59369

## What?

In the Site Editor > Design > Navigation > Any navigation menu, the labeling of the list view doesn't provide any informatiom about _what_ the item is. The only information provided is the link _type_.

## Why?

To be able to understand what they are going to edit, users need to know what the link actually is, ie. the link text.

## How?

- Provides a more meaningful label for the navigation-link blocks in the list view.
- Also simplifies the 'Options for %s' label of the ellipsis button to 'Actions', for simplicity and consistency.

## Testing Instructions




- Use a screen reader, for example VoiceOver with Safari.
- Go to the Site Editor > Design > Navigation > Any navigation menu.
- Navigate through the list view items in the navigation panel.
- Observe the items are now announced with their visible text so users can understand what the link they are going to edit is.
- Observe the item position is correctly provided in the visually hidden description of each item e.g. `Block 3 of 5, Level 1. `.
- Once on a selected item, use the arrow keys to move focus to the ellipsis button or hover the button to reveal its tooltip.
- Observe the button is announced as 'Actions' (same for the tooltip).
- Edit a navigation link, empty its label and save.
- ~Observe in the editor navigation panel list view the empty link has now a label 'Untitled'.~
- Update: The empty link will be empty in the list view, this is unchanged and will need to be addressed separately.
- Go to the post editor.
- Add a variety of blocks, for example:
  - blocks with default names
  - blocks with custom names (renamed by users)
  - synced patterns with user provided names
- Navigate the list view items with a screen reader and observe they are announced in the most meaningful way for that specific kind of block e.g.: custom name for renamed blocks, user provided name for synced patterns, default block name for default blocks etc.
- Add a Group block with a paragraph and lock the Group.
- In the list view, inspect the Group item and observe the visually hidden description provides both the block position and the locked state e.g.: `Block 3 of 5, Level 1. This block is locked.`
- Add a Heading block that only contains spaces. Save.
- Observe the heading is named 'Heading' in the List View.
- Onserve the headint is reported as empty heading in the Document Outline, screenshot:

![Screenshot 2024-03-19 at 09 00 11](https://github.com/WordPress/gutenberg/assets/1682452/641d6216-8f5d-4f34-b553-94656d2cd058)


### Testing Instructions for Keyboard


## Screenshots or screencast
